### ISSUE TITLE
[Java] CWE-016: Query to detect insecure configuration of Spring Boot Actuator 

### ISSUE THREAD

**Issue by luchua-bc** [github/securitylab#310]
## CVE ID(s)

*List the CVE ID(s) associated with this vulnerability. GitHub will automatically link CVE IDs to the [GitHub Advisory Database](https://github.com/advisories).*

## Report

*Describe the vulnerability. Provide any information you think will help GitHub assess the impact your query has on the open source community.*

Spring Boot is a popular framework that facilitates the development of stand-alone applications and micro services. Spring Boot Actuator helps to expose production-ready support features against Spring Boot applications.

Endpoints of Spring Boot Actuator allow to monitor and interact with a Spring Boot application. Exposing unprotected actuator endpoints through configuration files can lead to information disclosure or even remote code execution vulnerability.

Rather than programmatically permitting endpoint requests or enforcing access control, frequently developers simply leave management endpoints publicly accessible in the application configuration file <code>application.properties</code> without enforcing any access control through Spring Security.

This is a very common issue and is also one of the highest rewarded vulnerabilities on the HackerOne platform. The query detects this issue in Spring Boot projects with Maven and application.properties, which is the most widely adopted deployment scenario.

Relevant PR: [#5384](https://github.com/github/codeql/pull/5384)

- [x] Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc). *We would love to have you spread the word about the good work you are doing*

## Result(s)

*Provide at least one useful result found by your query, on some revision of a real project.*

- [Service Integration bora App](https://github.com/JeffMSantos/bora)

**Comment by ghsecuritylab** [comment]
Your submission is now in status **SecLab review**.

_For information, the evaluation workflow is the following:
CodeQL initial assessment > SecLab review > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by m-y-mo** [comment]
@luchua-bc We've ran your query on lgtm. However, the query does not flag any result. Would you be able to provide at least one result for this query on lgtm (or a local snapshot that you built if there's build problem etc. on lgtm) so that we can assess it? Thanks.

**Comment by luchua-bc** [comment]
@m-y-mo This is a very common issue and there shall hundreds if not thousands of projects with this issue.

The reason no result is flagged is that the query checks <code>application.properties</code>, which is a Java properties file that is not included in the CodeQL database by default.

To enable this, I added <code>export LGTM_INDEX_PROPERTIES_FILES=true</code> to my local bash <code>.bash_profile</code>. Is there a way to enable properties file inclusion on the LGTM platform?

@luchua-bc

**Comment by ghsecuritylab** [comment]
Your submission is now in status **SecLab finalize**.

_For information, the evaluation workflow is the following:
CodeQL initial assessment > SecLab review > CodeQL review > SecLab finalize > Pay > Closed_

**Comment by xcorail** [comment]
Hey @luchua-bc 

Thanks for the submission! We have reviewed your report and validated your findings. After internally assessing the findings and the query we have determined this query is not eligible for a reward under the Bug Bounty program for the following reason (from our official rules):

>To be eligible for a bounty, queries must be non-trivial, and meet a minimum complexity requirement. More concretely, queries that simply look for one or two AST elements, or that could be easily implemented with a linter or simple grep, may not be considered interesting enough for a bounty. Queries must also be sufficiently general to cover a class of vulnerabilities and may not be tailored to a specific CVE.
A good way to ensure that your queries meet this requirement is to ensure it uses some more advanced analysis, like data-flow or control-flow.

Even though this issue was assessed as being ineligible, we think it could find real security problems, and we appreciate the effort you put writing it, so we'll continue working on your PR, and we would still like to offer you $500.

Best regards and happy hacking!
